### PR TITLE
COMMERCE-11491, COMMERCE-11494

### PR DESCRIPTION
**Pull request by brianikim**
@andrea-ale-sbarra ,

https://liferay.atlassian.net/browse/COMMERCE-11494
https://liferay.atlassian.net/browse/COMMERCE-11491

As I mentioned on https://liferay.atlassian.net/browse/COMMERCE-11863, I implemented both ways to handle the NPEs. Let me know which one you prefer.

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java [hunk 1]
```diff
@@ -57,17 +60,35 @@ public ProductOption toDTO(DTOConverterContext dtoConverterContext)
...
 				skuContributor = cpDefinitionOptionRel.isSkuContributor();
+
+				setKey(
+					() -> {
+						if (cpOption != null) {
+							return cpOption.getKey();
+						}
+
+						return StringPool.BLANK;
+					});
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java:75]:
```suggestion
					key = cpDefinitionOptionRel.getKey();
```


This was wrong in the beginning we should return the saved key for the CPOption, this is the most important field of the CPOption feature

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java [hunk 2]
```diff
@@ -57,17 +60,35 @@ public ProductOption toDTO(DTOConverterContext dtoConverterContext)
...
+				setOptionId(
+					() -> {
+						if (cpOption != null) {
+							return cpOption.getCPOptionId();
+						}
+
+						return null;
+					});
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java:83]:
```suggestion
						if (cpOption == null) {
							return null;
						}

						return cpOption.getCPOptionId();
					});
```

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Forget about this I'll try this in my local and will resend another fix for this.

Thank you!

**Comment by brianikim** [comment]
ci:close

**Pull request closed without merging**